src/main/java/org/soen343/controllers/DashboardController.java

```diff
@@ -53,6 +53,8 @@ public void initialize() {
         SimulationContextService.getInstance().attachObserver(houseLayoutController).attachObserver(simulationContextController).attachObserver(simulationInfoController).attachObserver(smartHomeSimulatorModuleController);
         SHSModule.getInstance().attachObserver(houseLayoutController).attachObserver(simulationContextController).attachObserver(simulationInfoController).attachObserver(smartHomeSimulatorModuleController);
         SHCModule.getInstance().attachObserver(houseLayoutController).attachObserver(simulationContextController).attachObserver(simulationInfoController).attachObserver(smartHomeSimulatorModuleController);
+        DashboardService.getInstance().attachObserver(simulationContextController).attachObserver(smartHomeSimulatorModuleController);
+        DashboardService.getInstance().attachObserver(simulationContextController).attachObserver(smartHomeCoreModuleController);
```

Comment:
You dont need this line. 

You chain call the attachObserver(). So just add smartHomeCoreModuleController at the end of the last line